libs/x-content/src/main/java/org/elasticsearch/common/xcontent/AbstractObjectParser.java

```diff
@@ -294,6 +295,23 @@ public void declareRequiredFieldSet(String... requiredSet) {
         this.requiredFieldSets.add(requiredSet);
     }
 
+    /**
+     * Declares a set of fields of which at most one must appear for parsing to succeed
+     *
+     * E.g. <code>declareExclusiveFieldSet("foo", "bar");</code> means that only one of 'foo'
+     * or 'bar' must be present, and if both appear then an exception will be thrown.
+     *
+     * Multiple exclusive sets may be declared
```

Comment:
I think it might be worth mentioning that it doesn't make `foo` or `bar` required.

Reply:
++